movr.py

```diff
@@ -113,256 +123,205 @@ def add_vehicle_helper(session, city, owner_id, current_location, type, vehicle_
-        queries_to_run["fk_alters"].append("DROP INDEX IF EXISTS vehicles_city_idx;")
-        queries_to_run["fk_alters"].append(
-            "ALTER TABLE vehicles ADD CONSTRAINT fk_owner_id_ref_users_mr FOREIGN KEY (city, owner_id) REFERENCES users (city,id);")
-
-        # rides
-        queries_to_run["fk_alters"].append("ALTER TABLE rides DROP CONSTRAINT fk_rider_id_ref_users;")
-        queries_to_run["fk_alters"].append("CREATE INDEX ON rides (city, rider_id);")
-        queries_to_run["fk_alters"].append(
-            "ALTER TABLE rides ADD CONSTRAINT fk_rider_id_ref_users_mr FOREIGN KEY (city, rider_id) REFERENCES users (city,id);")
-        queries_to_run["fk_alters"].append("ALTER TABLE rides DROP CONSTRAINT fk_vehicle_id_ref_vehicles;")
-        queries_to_run["fk_alters"].append("CREATE INDEX ON rides (city, vehicle_id);")
-        queries_to_run["fk_alters"].append(
-            "ALTER TABLE rides ADD CONSTRAINT fk_vehicle_id_ref_vehicles_mr FOREIGN KEY (city, vehicle_id) REFERENCES vehicles (city,id);")
-        queries_to_run["fk_alters"].append("DROP INDEX IF EXISTS rides_auto_index_fk_rider_id_ref_users;")
-        queries_to_run["fk_alters"].append("DROP INDEX IF EXISTS rides_auto_index_fk_vehicle_id_ref_vehicles;")
-
-
-        # user_promo_codes
-        queries_to_run["fk_alters"].append("ALTER TABLE user_promo_codes DROP CONSTRAINT fk_user_id_ref_users;")
-        queries_to_run["fk_alters"].append(
-            "ALTER TABLE user_promo_codes ADD CONSTRAINT fk_user_id_ref_users_mr FOREIGN KEY (city, user_id) REFERENCES users (city,id);")
-
-
-        # drop all the old pks that became unique indexes
-        queries_to_run["fk_alters"].append("DROP INDEX IF EXISTS users_id_key CASCADE;")
-        queries_to_run["fk_alters"].append("DROP INDEX IF EXISTS user_promo_codes_user_id_code_key CASCADE;")
-        queries_to_run["fk_alters"].append("DROP INDEX IF EXISTS rides_id_key CASCADE;")
-        queries_to_run["fk_alters"].append("DROP INDEX IF EXISTS vehicles_id_key CASCADE;")
-        queries_to_run["fk_alters"].append("DROP INDEX IF EXISTS vehicle_location_histories_ride_id_timestamp_key CASCADE;")
-
-        return queries_to_run
+        # This function uses raw SQL, as there is no functional mapping to multi-region ALTER statements, in any ORM/tool
+        # We should really execute all schema changes with Alembic (the migration tool built for SQLAlchemy), but this should work for now
+        # See https://docs.sqlalchemy.org/en/14/core/metadata.html#altering-database-objects-through-migrations
+
+        # Alter database statements
+        region_tups = self.session.execute(
+            text('SELECT region FROM [SHOW REGIONS]')).fetchall()
+        regions = []
+        for tup in region_tups:
+            regions.append(tup[0])
+
+        if self.primary_region is None:
+            self.primary_region = regions[0]
+
+        add_primary_region_query = 'ALTER DATABASE movr PRIMARY REGION "{0}"'.format(
```

Comment:
the database name should be dynamic. This is coming from the connection string, which might not point to a db named "movr". See https://github.com/cockroachdb/movr/blob/multi-region-ux/movr.py#L197

Reply:
Done.